R/qenv-constructor.R

```diff
@@ -24,13 +24,7 @@ setMethod(
   "new_qenv",
   signature = c(env = "environment", code = "expression"),
   function(env, code) {
-    new_env <- rlang::env_clone(env, parent = parent.env(.GlobalEnv))
-    lockEnvironment(new_env, bindings = TRUE)
-    id <- sample.int(.Machine$integer.max, size = length(code))
-    methods::new(
-      "qenv",
-      env = new_env, code = code, warnings = rep("", length(code)), messages = rep("", length(code)), id = id
-    )
+    new_qenv(env, format_expression(code))
```

Comment:
let's move `format_expression` to `get_code`. Stored code can be "dirty" in favour of faster constructor

Reply:
make sense!